honeybee_radiance/primitive/material/plastic.py

```diff
@@ -0,0 +1,259 @@
+
+        Usage:
+            wall_material = Plastic("generic_wall", .55, .65, .75)
+            print(wall_material)
+        """
+        Material.__init__(self, name, modifier=modifier,
+                          dependencies=dependencies)
+        self.r_reflectance = r_reflectance
+        self.g_reflectance = g_reflectance
+        self.b_reflectance = b_reflectance
+        self.specularity = specularity
+        self.roughness = roughness
+
+        self._update_values()
+
+    def _update_values(self):
+        "update value dictionaries."
+        self._values[2] = [
+            self.r_reflectance, self.g_reflectance, self.b_reflectance,
+            self.specularity, self.roughness
+        ]
+
+        if self.specularity > 0.1:
+            print("Warning: Specularity values above .1 is uncommon for plastic.")
+        if self.roughness > 0.2:
+            print("Warning: Roughness values above .2 is uncommon.")
+
+    @property
+    def r_reflectance(self):
+        """Transmissivity for red channel.
+
+        The value should be between 0 and 1 (Default: 0).
+        """
+        return self._r_reflectance
+
+    @r_reflectance.setter
+    def r_reflectance(self, reflectance):
+        self._r_reflectance = typing.float_in_range(reflectance, 0, 1)
+
+    @property
+    def g_reflectance(self):
+        """Transmissivity for green channel.
+
+        The value should be between 0 and 1 (Default: 0).
+        """
+        return self._g_reflectance
+
+    @g_reflectance.setter
+    def g_reflectance(self, reflectance):
+        self._g_reflectance = typing.float_in_range(reflectance, 0, 1)
+
+    @property
+    def b_reflectance(self):
+        """Transmissivity for blue channel.
+
+        The value should be between 0 and 1 (Default: 0).
+        """
+        return self._b_reflectance
+
+    @b_reflectance.setter
+    def b_reflectance(self, reflectance):
+        self._b_reflectance = typing.float_in_range(reflectance, 0, 1)
+
+    @property
+    def specularity(self):
+        """Fraction of specularity.
+
+        In most cases specularity fractions greater than 0.1 are not realistic
+        (Default: 0).
+        """
+        return self._specularity
+
+    @specularity.setter
+    def specularity(self, spec_value):
+        self._specularity = typing.float_in_range(spec_value, 0, 1)
+
+    @property
+    def roughness(self):
+        """Roughness is specified as the rms slope of surface facets.
+
+        A value of 0 corresponds to a perfectly smooth surface, and a value of 1
+        would be a very rough surface. Roughness values greater than 0.2 are not
+        very realistic. (Default: 0).
+        """
+        return self._roughness
+
+    @roughness.setter
+    def roughness(self, roughness_values):
+        self._roughness = typing.float_in_range(roughness_values, 0, 1)
+
+    @property
+    def average_reflectance(self):
+        """Calculate average reflectance of plastic material."""
```

Comment:
See the link that I shared up there. If we can make a short version of that logic I agree that it is good to add it here otherwise one can google it and find the link that I shared up there.